### DISCUSSION TITLE
Speed Measurement

### DISCUSSION THREAD

**Discussion by mwareman** [blakeblackshear/frigate#15264] [Ideas]
As an idea, 💡 if we were to measure and mark distances based on the view of the camera (for instance, draw a line along a road and physically measure it on the ground, and document the length in Frigate) - could this later be used to estimate the speed of obtects moving thru the image? The idea would be to measure the speed along the line. Multiple lines could be defined - and when a single object is detected by multiple lines the highest measured speed reported with the object.

I don't care to capture all cars moving along the street. Burt, card moving faster than a given speed I would like to create 'alerts'. 

Is this possible to infer?

**Comment by NickM-27** [comment]
Yes, it would be possible to infer speed with this data

**Reply by mwareman** [reply]
Awesome! I would love each detection to have a speed attribute added when it's sent to Home Assistant and MQTT, and being able to optionally configure a speed value on object types above which will cause a detection.
